Editor/LyumaAv3EditorSupport.cs

```diff
@@ -221,9 +223,46 @@ public static class LyumaAv3EditorSupport
 
 		public static void OnPlayModeStateChange(UnityEditor.PlayModeStateChange pmsc) {
 			// We don't want any of our callbacks causing trouble outside of play mode.
-			if (pmsc != UnityEditor.PlayModeStateChange.EnteredPlayMode) {
+			if (pmsc != UnityEditor.PlayModeStateChange.EnteredPlayMode)
+			{
 				CompilationPipeline.assemblyCompilationStarted -= WorkaroundDestroyManagersBeforeCompile;
 			}
+
+			// Writing all disabled objects into SessionState
+			if (pmsc == PlayModeStateChange.ExitingEditMode)
+			{
+				List<GameObject> avatars = GameObject.FindObjectsOfType<VRCAvatarDescriptor>()
+					.Select(x => x.gameObject)
+					.Where(x => x.activeSelf).ToList();
+
+				bool emulatorEnabled = false;
+				
+				foreach (var lyumaAv3Emulator in GameObject.FindObjectsOfType<LyumaAv3Emulator>())
+				{
+					emulatorEnabled = true;
+					lyumaAv3Emulator.AvatarList = avatars;
+				}
+
+				if (emulatorEnabled)
+				{
+					foreach (var gameObject in avatars)
+					{
+						gameObject.SetActive(false);
+					}
+				}
+			}
+
+			// Obtaining disabled objects from SessionState and enabling them
+			if (pmsc == PlayModeStateChange.EnteredEditMode)
+			{
+				foreach (var lyumaAv3Emulator in GameObject.FindObjectsOfType<LyumaAv3Emulator>())
+				{
+					foreach (var gameObject in lyumaAv3Emulator.AvatarList)
+					{
+						gameObject.SetActive(true);
+					}
+				}
+			}
```

Comment:
I would feel better if this also clears the AvatarList, so it doesn't change between exit and enter edit mode, so it's always empty during edit mode.